socketshark/__init__.py

```diff
@@ -111,7 +111,8 @@ def signal_shutdown(self):
         self.metrics.set_ready(False)
 
     async def _redis_connection_handler(self):
-        await self.redis.wait_closed()
+        tasks = [c.redis.wait_closed() for c in self.redis_connections]
+        await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
```

Comment:
This function is meant to restart the service in case a Redis connection is closed. So in this case we want to know if any one connection has ended, rather than stay alive with partial functionality.

Reply:
Ah, gotcha. I think this isn't obvious at all. Perhaps we can add some comments / docstrings?

Reply:
My understanding of this function is it guards against an unexpected Redis disconnect. If one connection is lost, the service will not function correctly. Going through the full shutdown procedure tells the client Webscoket connections to attempt to reconnect to another host. If we terminate abruptly there is a risk of clients losing subscriptions.

I'll be adding a docstring per Stefan's recommendation here.